domain/reflection/pattern_detection.py

```
@@ -0,0 +1,525 @@
+                confidence=0.8,
+            )
+            self.patterns[pattern_id] = pattern
+            
+        self.logger.info(f"Initialized {len(common_patterns)} common patterns")
+
+    def _load_patterns(self) -> None:
+        """Load patterns from storage."""
+        try:
+            if self.storage_path.exists():
+                with open(self.storage_path, 'rb') as f:
+                    data = pickle.load(f)
```

Ruff (0.13.1)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)